### PR TITLE
complete error type of `len` , `for` function

### PR DESCRIPTION
**Pull request by garry911128**
complete the len(1) and for x in `int or bool or none` type will throw run time error

Compilation:
Compilation: 23/48 : 47%
Code behavior: 23/48 : 47%
Expected output: 19/48 : 39%

### CODE REVIEW

File: mini-python-ocaml/compile.ml [hunk 1]
```diff
@@ -209,8 +234,24 @@ let rec compile_expr (env: env_t) (expr: Ast.texpr) : X86_64.text * X86_64.data
...
+            failwith "Implement len logic here"
+          | `int ->
+            call "runtime_error", nop, `int
+          | _ -> (* 錯誤類型 *)
+            failwith "TypeError: object of type is not iterable"
+          end
+        | _ ->
+          failwith "TypeError: len() takes exactly one argument"
+      else
+        failwith "Unsupported function call"
```
Comment (kyynk) [inline mini-python-ocaml/compile.ml]:
```suggestion
  | TEcall (fn, args) ->
    if fn.fn_name = "len" then
      match args with
      | [e] ->
        let _, _, arg_type = compile_expr env e in
        begin match arg_type with
        | `string | `list -> (* 正確的類型 *)
          (* len 的實現邏輯 *)
          failwith "Implement len logic here"
        | `int ->
          call "runtime_error", nop, `int
        | _ -> (* 錯誤類型 *)
          failwith "TypeError: object of type is not iterable"
        end
      | _ ->
        failwith "TypeError: len() takes exactly one argument"
    else
      failwith "Unsupported function call"
```

**Review by kyynk — CHANGES_REQUESTED** [review]

**Comment by Vghxv** [comment]
remove Chinese comments

File: mini-python-ocaml/test.py [hunk 2]
```diff
@@ -1,3 +1 @@
-
-g = ["yee", 2, True, 4]
-print(g)
\ No newline at end of file
+print("make" + "love")
```
Comment (Vghxv) [inline mini-python-ocaml/test.py]: again then I quit

File: mini-python-ocaml/compile.ml [hunk 3]
```diff
@@ -56,44 +56,69 @@ let rec compile_expr (env: env_t) (expr: Ast.texpr) : X86_64.text * X86_64.data
...
+          call "strcpy_wrapper" ++
+    
+          (* 拼接第二個字符串 *)
+          movq (reg rsi) (reg rdi) ++   (* 新分配內存地址 -> rdi *)
+          movq (reg rsi) (reg rsi) ++   (* 第二個字符串地址 -> rsi *)
+          call "strcat_wrapper" ++
+          movq (reg rdi) (ind rax)      (* 保存結果地址 *)
+        in
+        concat_code, nop, `string
+
```
Comment (Vghxv) [inline mini-python-ocaml/compile.ml]: didn't consider len and tag of concatenated string
btw I did this

**Comment by Vghxv** [comment]
and please stop merging main into another branch next time, use rebase instead

**Review by Vghxv — APPROVED** [review]
LGTM

**Review by kyynk — APPROVED** [review]
LGTM

**Pull request merged**